packages/core/PluginLoader.ts

```diff
@@ -61,35 +125,124 @@ export default class PluginLoader {
     )
   }
 
-  async loadPlugin(definition: PluginDefinition) {
-    const parsedUrl = new URL(definition.url)
-    if (
-      !parsedUrl.protocol ||
-      parsedUrl.protocol === 'http:' ||
-      parsedUrl.protocol === 'https:'
-    ) {
-      await this.loadScript(definition.url)
-      const moduleName = definition.name
-      const umdName = `JBrowsePlugin${moduleName}`
-      // Based on window-or-global
-      // https://github.com/purposeindustries/window-or-global/blob/322abc71de0010c9e5d9d0729df40959e1ef8775/lib/index.js
-      const scope =
-        (typeof self === 'object' && self.self === self && self) ||
-        (typeof global === 'object' && global.global === global && global) ||
-        this
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const plugin = (scope as any)[umdName] as { default: PluginConstructor }
-      if (!plugin) {
-        throw new Error(
-          `plugin ${moduleName} failed to load, ${scope.constructor.name}.${umdName} is undefined`,
-        )
-      }
-
-      return plugin.default
+  async loadCJSPlugin(
+    pluginDefinition: CJSPluginDefinition,
+  ): Promise<LoadedPlugin> {
+    const parsedUrl = new URL(pluginDefinition.cjsUrl)
+    if (parsedUrl.protocol !== 'http:' && parsedUrl.protocol !== 'https:') {
+      throw new Error(
+        `cannot load plugins using protocol "${parsedUrl.protocol}"`,
+      )
     }
-    throw new Error(
-      `cannot load plugins using protocol "${parsedUrl.protocol}"`,
+    const fs: typeof import('fs') = require('fs')
+    const path: typeof import('path') = require('path')
+    const os: typeof import('os') = require('os')
+    const http: typeof import('http') = require('http')
+    const fsPromises = fs.promises
+    // On macOS `os.tmpdir()` returns the path to a symlink, see:
+    // https://github.com/nodejs/node/issues/11422
+    const systemTmp = await fsPromises.realpath(os.tmpdir())
+    const tmpDir = await fsPromises.mkdtemp(
+      path.join(systemTmp, 'jbrowse-plugin-'),
     )
+    let plugin: LoadedPlugin | undefined = undefined
+    try {
+      const pluginLocation = path.join(tmpDir, sanitize(parsedUrl.href))
+      const pluginLocationRelative = path.relative('.', pluginLocation)
+
+      const pluginDownload = new Promise<void>((resolve, reject) => {
+        const file = fs.createWriteStream(pluginLocation)
+        http
```

Comment:
random not-sure-if-needed things but could use follow-redirects instead of raw http module, and might want to toggle between http and https depending on protocol(?)